File: file-tree-walker.js
(function () {
  "use strict";

  var fs = require("fs");
  var path = require("path");

  var walkFileTree = function walkFileTree (f, proc) {
    if (fs.statSync(f).isDirectory()) {
      fs.readdir(f, function (err, files) {
        if (err) {
          throw err;
        }
        files.map(function (file) {
          walkFileTree(path.join(f, file), proc);
        });
      });
    } else {
      proc(f);
    }
  };

  var FileTreeWalker =  {
    walkFileTree: walkFileTree
  };
  module.exports = FileTreeWalker;

  FileTreeWalker.walkFileTree("C:\\", function (f) {
    console.log(path.resolve(f));
  });
}.call(this));

File: file_tree_walker.py
import os


def walk_file_tree(f, proc):
    for root, dirs, files in os.walk(f):
        for file in files:
            proc(os.path.join(root, file))


if __name__ == "__main__":
    walk_file_tree("C:\\", lambda f: print(os.path.abspath(f)))
